ConsumerEdition/DragonBoard-410c/Guides/AOSP.md

```diff
@@ -5,63 +5,78 @@ redirect_from: /db410c-getting-started/Guides/AOSP.md/
+```
 Then run:
-
-    repo init -u https://android.googlesource.com/platform/manifest -b master
-    cd .repo
-    git clone https://github.com/robherring/android_manifest.git -b master local_manifests
-    cd ..
-    repo sync -j24
-
+```shell
+repo init -u https://android.googlesource.com/platform/manifest -b master
+cd .repo
+git clone https://github.com/robherring/android_manifest.git -b master local_manifests
+cd ..
+repo sync -j24
+```
 It might take quite a bit of time to fetch the entire AOSP source code!
 
-# Firmware files
+# Firmware files (Optional)
 
 Download the firmware setup package from here: https://drive.google.com/open?id=0B2zT38Egh-1TVkZoYUliOXdxaUk, and run the following command to prepare the firmware files in your Android environment:
-
-    tar -xf qcom-db410c-NOU.tgz
-    ./extract-qcom-db410c.sh
-
+```shell
+tar -xf qcom-db410c-NOU.tgz
+./extract-qcom-db410c.sh
+```
 When needed, hit yes, to agree to license.
 
 Note that using this script is a temporary solution until the proper firmware package is published on 96boards website.
 
-# Building
-
-    source build/envsetup.sh
-    lunch db410c
-    make -j24
+# Building AOSP
+```shell
+source build/envsetup.sh
+lunch linaro_arm64-userdebug
+make -j24
+```
+
+# Building Kernel
+```shell
+cd ..
+git clone https://git.kernel.org/pub/scm/linux/kernel/git/robh/linux.git -b qcom-4.14 <KERNEL_TOP>
+cd <KERNEL_TOP>
+make ARCH=arm64 db410c_android_defconfig android-base.config android-recommended.config
+make CROSS_COMPILE=aarch64-android-gnu- ARCH=arm64 all
+cat arch/arm64/boot/Image.gz arch/arm64/boot/dts/qcom/apq8016-sbc.dtb > Image.gz-dtb
+mkbootimg --kernel Image.gz-dtb --ramdisk <ANDROID_TOP>/out/target/product/linaro_arm64/ramdisk.img --output boot-db410c.img --pagesize 2048 --base 0x80000000 --cmdline 'androidboot.selinux=permissive firmware_class.path=/system/vendor/firmware/'
+```
 
 # Flashing the board
 
 While holding the S4 button, power the DragonBoard 410c board to enter fastboot mode. Before flashing the Android images, we need to flash the bootloaders and GPT. e.g. run the following commands:
-
-    mkdir ~/db410-bootlaoder
-    cd ~/db410-bootlaoder
-    wget http://builds.96boards.org/snapshots/dragonboard410c/linaro/rescue/75/dragonboard410c_bootloader_emmc_aosp-75.zip
-    unzip dragonboard410c_bootloader_emmc_aosp-75.zip
-    ./flashall
-
+```shell
+cd ..
+mkdir ~/db410-bootlaoder
+cd ~/db410-bootlaoder
+wget http://builds.96boards.org/snapshots/dragonboard410c/linaro/rescue/latest/dragonboard410c_bootloader_emmc_aosp-*.zip
+unzip dragonboard410c_bootloader_emmc_aosp-*.zip
+./flashall
+```
 We can now flash the Android images.
-
-    cd <ANDROID_TOP>
-    fastboot flash boot out/target/product/db410c/boot.img
-    fastboot flash system out/target/product/db410c/system.img
-    fastboot flash cache out/target/product/db410c/cache.img
-    fastboot flash userdata out/target/product/db410c/userdata.img
-    fastboot reboot
-
+```shell
+cd ..
+cd <KERNEL_TOP>
```

Comment:
this does not look right. you either need to create the boot image in ANDROID_TOP/out/target/... or in KERNEL_TOP/bootxx.img

Reply:
doh! just a sec